### PR TITLE
Add PermissionUtilsTest for findHighestPermissionValue

### PR DESCRIPTION
**Pull request by Nrleryxx**
Add unit tests for PermissionUtils.findHighestPermissionValue covering both the permission map and hasPermission fallback paths including validation edge cases

### CODE REVIEW

File: proxy/src/test/java/com/velocityctd/proxy/permission/PermissionUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        () -> PermissionUtils.findHighestPermissionValue(subject, "perm.", 0));
+    assertThrows(IllegalArgumentException.class,
+        () -> PermissionUtils.findHighestPermissionValue(subject, "perm.", -1));
+  }
+
+  @Test
+  void returnsEmptyWhenPermissionMapIsEmptyAndFastTrackFails() {
+    PermissionSubject subject = mock(PermissionSubject.class);
+    when(subject.getPermissionMap()).thenReturn(Map.of());
+    when(subject.getPermissionValue("perm.100")).thenReturn(Tristate.FALSE);
```
Comment (WouterGritter) [inline proxy/src/test/java/com/velocityctd/proxy/permission/PermissionUtilsTest.java]: Instead of hard-coding just "perm.100" here, it's probably better to mock the PermissionSubject to return false for every permission with an empty permission map (or the middle tristate, unset i beleive)

File: proxy/src/test/java/com/velocityctd/proxy/permission/PermissionUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    assertEquals(Optional.empty(),
+        PermissionUtils.findHighestPermissionValue(subject, "perm.", 100));
+  }
+
+  @Test
+  void returnsMaxFromPermissionMap() {
+    PermissionSubject subject = mock(PermissionSubject.class);
+    when(subject.getPermissionMap()).thenReturn(Map.of("timeout.30", true));
+    when(subject.getPermissionValue("timeout.100")).thenReturn(Tristate.FALSE);
```
Comment (WouterGritter) [inline proxy/src/test/java/com/velocityctd/proxy/permission/PermissionUtilsTest.java]: Should the mock also return true for `subject.getPermissionValue("timeout.30")` here? That would be mocking the behavior of a real permission subject. Maybe a helper method in the test class to set up a mock PermissionSubject would be handy here.

Thinking about it, mocking it might even be too complex, could we use a `class TestPermissionSubject implements PermissionSubject` here? Pass in exactly which permissions it has, and a flag for returning null on `getPermissionMap()`? Just thinking out loud here, could be too verbose.

**Review by WouterGritter — APPROVED** [review]
